samples/bluetooth/bap_broadcast_sink/src/main.c

```diff
@@ -281,83 +289,115 @@ static bool subgroup_get_valid_bis_indexes_cb(const struct bt_bap_base_subgroup
 							      ? subgroup_chan_allocation
 							      : BT_AUDIO_LOCATION_MONO_AUDIO;
 		}
+		/* keep allocation for later use */
+		bis_index->audio_allocation[data.bis[i].index - 1] = data.bis[i].chan_allocation;
+		printk("bis_index->audio_allocation[%u] = 0x%08x\n", data.bis[i].index - 1,
+		       data.bis[i].chan_allocation);
 	}
 
 	/* Get the BIS indexes */
 	for (uint8_t i = 0U; i < data.cnt; i++) {
 		if ((data.bis[i].chan_allocation == CONFIG_TARGET_BROADCAST_CHANNEL) ||
-		    ((data.bis[i].chan_allocation & CONFIG_TARGET_BROADCAST_CHANNEL) ==
-		     CONFIG_TARGET_BROADCAST_CHANNEL)) {
-			/* Exact match */
+		    (CONFIG_TARGET_BROADCAST_CHANNEL == BT_AUDIO_LOCATION_MONO_AUDIO)) {
+			/* Exact match (accept all allocations as mono) */
 			bis_indexes = BT_ISO_BIS_INDEX_BIT(data.bis[i].index);
 
-			printk("Channel allocation match. BIS index bitfield 0x%x\n", bis_indexes);
-			*(uint32_t *)user_data = bis_indexes;
-
-			return false; /* bis index found */
+			printk("Channel allocation match, exact. BIS idx bitfield 0x%x\n",
+			       bis_indexes);
+			bis_index->bis_index_bitfield[bis_index->num_subgroups] = bis_indexes;
+			break; /* no need to go futher */
```

Comment:
```suggestion
			break; /* no need to go futher */
```

What if one of the other BIS in this subgroup is `BT_AUDIO_LOCATION_MONO_AUDIO`? Isn't that a better choice then?